### PR TITLE
Oracle turns off auto commit to drop temporary table

### PR DESCRIPTION
**Pull request by mwd410**
## Description

This is hopefully the final battle with Oracle to drop our temporary tables correctly, related to https://github.com/trinodb/trino/issues/16277

In my last PR, I forgot about the whole concept of auto commit. The result was the lock we acquired before the drop table was immediately released and had no effect.

## Release notes

(X) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java [hunk 1]
```diff
@@ -302,6 +302,7 @@ protected void dropTable(ConnectorSession session, RemoteTableName remoteTableNa
         String dropTableSql = "DROP TABLE " + quotedTable;
         try (Connection connection = connectionFactory.openConnection(session)) {
             if (temporaryTable) {
+                connection.setAutoCommit(false);
```
Comment (losipiuk) [inline plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java:305]: what is the motivation behind the change? Should we commit explicitly then? Or was switching auto-commit back to true just before connection is closed for that?

Comment (losipiuk) [reply plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java:305]: image

Comment (losipiuk) [reply plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java:305]: I see:

image

Comment (losipiuk) [reply plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java:305]: Please add a comment below - or just call `commit()` - will be more readable I think

Comment (mwd410) [reply plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java:305]: The motivation is that, because autoCommit is true when we get the Connection, the lock is immediately released because that `LOCK TABLE` statement is auto-committed, so the Connection has to re-acquire the lock for the `DROP TABLE`, so we have the same situation as we did before. Turning off auto commit holds onto that lock until after the drop table. Oracle's docs say DDL always auto-commits, but I reset autoCommit to true after the drop just in case.

Comment (mwd410) [reply plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java:305]: Oh, weird - when i typed out my reply, i only had your first comment in this thread visible to me.

I think i want to keep it as `setAutoCommit(true)` as it properly handles both cases: 
image

**Review by losipiuk — APPROVED** [review]
